frontend/src/concepts/pipelines/utils.ts

```diff
@@ -1,10 +1,27 @@
-import { deletePipelineCR, deleteSecret } from '~/api';
-import { ExternalDatabaseSecret } from '~/concepts/pipelines/content/configurePipelinesServer/const';
+import { deletePipelineCR, deleteSecret, getPipelinesCR } from '~/api';
+import {
+  DSPA_SECRET_NAME,
+  ExternalDatabaseSecret,
+} from '~/concepts/pipelines/content/configurePipelinesServer/const';
 import { ELYRA_SECRET_NAME } from '~/concepts/pipelines/elyra/const';
 
-export const deleteServer = (namespace: string): Promise<void> =>
-  Promise.allSettled([
+export const deleteServer = async (namespace: string): Promise<void> => {
+  const dspa = await getPipelinesCR(namespace);
+  const dspaSecretName =
+    dspa.spec.objectStorage.externalStorage?.s3CredentialsSecret.secretName ?? '';
+  return Promise.allSettled([
     deleteSecret(namespace, ExternalDatabaseSecret.NAME),
     deleteSecret(namespace, ELYRA_SECRET_NAME),
+    deleteSecret(namespace, DSPA_SECRET_NAME),
+    isGeneratedDSPAExternalStorageSecret(dspaSecretName) && deleteSecret(namespace, dspaSecretName),
     deletePipelineCR(namespace),
   ]).then(() => undefined);
+};
+
+/**
+ * Whether the Secret resource is generated
+ * If it's generated, it should be `secret-xxxxxx` or `dspa-secret`
+ * Use this test to avoid deleting the user-created secret for the pipeline server
+ */
+export const isGeneratedDSPAExternalStorageSecret = (name: string): boolean =>
```

Comment:
im guessing there is not a good way to differentiate between a user generated secret that was named to match how we name things. other than possibly adding some annotations

Reply:
Problem is the existing ones were not deleted and don't have additional information like that to key on (at least not that I saw)... so this is a wild shot in the dark to hopefully be somewhat able to clean things up.